<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Analysis - SARS-CoV-2 GTG lab documentation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Analysis";
    var mkdocs_page_input_path = "analysis.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> SARS-CoV-2 GTG lab documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../protocols/">Protocols</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../seq/">Sequencing</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../install/">Install</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../realtime/">Real-time monitoring</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Analysis</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#analysis">Analysis</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#nanopolish">Nanopolish</a></li>
        
            <li><a class="toctree-l3" href="#medaka">Medaka</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../tute/">Tutorial</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../inter/">interARTIC</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">SARS-CoV-2 GTG lab documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Analysis</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="analysis">Analysis</h1>
<p>The analysis of the output to attain a consensus genome for each sample, as well as find the lineage information of each sample, is done using the ARTIC bioinformatics pipeline and pangolin.</p>
<p>The are 2 methods within the ARTIC pipeline.</p>
<ol>
<li>Nanopolish (uses both fastq and fast5 files)</li>
<li>Medaka (uses only fastq files)</li>
</ol>
<p>We mostly use medaka for our routine samples. Either method works well enough for contact tracing and lineage analysis.</p>
<h2 id="nanopolish">Nanopolish</h2>
<h3 id="basic-pipeline">Basic pipeline</h3>
<h4 id="gather">Gather</h4>
<p>Collects the reads which pass quality and length filtering. The <code>--min-length/--max-length</code> flags depend on if you used the ligation kit or rapid kit.</p>
<h3 id="single-sample">Single sample</h3>
<h3 id="multiple-samples">Multiple samples</h3>
<h2 id="medaka">Medaka</h2>
<h3 id="basic-pipeline_1">Basic pipeline</h3>
<h3 id="single-sample_1">Single sample</h3>
<h3 id="multiple-samples_1">Multiple samples</h3>
<p>folder structure</p>
<pre><code>PKRP008209
├── base
├── medaka
├── nanopolish
└── PKRP008209
    └── 1905-8
        └── 20210113_0459_1C_PAF26262_d0402aba
            ├── fastq_fail
            └── fastq_pass
</code></pre>
<p>Make folders</p>
<pre><code>mkdir base medaka nanopolish
</code></pre>
<p>in the base folder</p>
<pre><code>cd base
</code></pre>
<p>Set up variable for run prefix (in this example, it is PKRP008209)</p>
<pre><code>RUN=PKRP008209
</code></pre>
<h4 id="gather_1">Gather</h4>
<p>Collects the reads which pass quality and length filtering. The <code>--min-length/--max-length</code> flags depend on if you used the ligation kit or rapid kit.</p>
<h4 id="ligation">Ligation</h4>
<p><strong>ARTIC 400bp primers</strong></p>
<pre><code>artic gather --min-length 300 --max-length 600 --prefix ${RUN} --directory ../${RUN}/*/ --no-fast5s
</code></pre>
<p><strong>Midnight 1200bp primers</strong></p>
<pre><code>artic gather --min-length 1000 --max-length 1400 --prefix ${RUN} --directory ../${RUN}/*/ --no-fast5s
</code></pre>
<p><strong>Eden/Kirby 2500bp primers</strong></p>
<pre><code>artic gather --min-length 2300 --max-length 2700 --prefix ${RUN} --directory ../${RUN}/*/ --no-fast5s
</code></pre>
<h4 id="rapid">Rapid</h4>
<p><strong>ARTIC 400bp primers</strong></p>
<pre><code>Probably not a good idea to use the rapid kit on 400bp amplicons, as it will make many fragments shorter then 300bp. The basecallers struggle with sequences smaller than this.
</code></pre>
<p><strong>Midnight 1200bp primers</strong></p>
<pre><code>artic gather --min-length 300 --max-length 1400 --prefix ${RUN} --directory ../${RUN}/*/ --no-fast5s
</code></pre>
<p><strong>Eden/Kirby 2500bp primers</strong></p>
<pre><code>artic gather --min-length 300 --max-length 2700 --prefix ${RUN} --directory ../${RUN}/*/ --no-fast5s
</code></pre>
<h4 id="demultiplexing">Demultiplexing</h4>
<h4 id="ligation_1">Ligation</h4>
<pre><code>artic demultiplex --threads 8 ${RUN}_fastq_pass.fastq
</code></pre>
<h4 id="rapid_1">Rapid</h4>
<pre><code>porechop --verbosity 2 --untrimmed -i "${RUN}_fastq_pass.fastq" -b ./ --rapid_barcodes --discard_middle --barcode_threshold 80 --threads 8 --check_reads 10000 --barcode_diff 5 &gt; ${RUN}_fastq_pass.fastq.demultiplexreport.txt

# it wont name the files properly this way, so rename them with this

for file in BC*.fastq; do mv $file ${RUN}_fastq_pass-${file}; done
</code></pre>
<h4 id="consensus-with-medaka">consensus with medaka</h4>
<p>using a for loop, we can name each folder and prefix each file with the sample information. Using the syntax: <code>for i in "barcode_number&lt;space&gt;sample_ID"&lt;space&gt;"..."</code></p>
<p>This will then take the fastq for each barcode_number, 01, 02, etc., and name it with the corresponding sample name, for example, if the sample name is R_nCoV_0013, then the command has the static portion <code>R_nCoV_${a[1]}</code>, and inserts the variable portion with the number, say, 0013. The commands below are for 2 barcoded samples, from barcodes 01 and 02, with their corresponding sample numbers, 0013, 0014. Change this part in the for loop to match the samples you wish to analyse.</p>
<p>for the barcodes, if you used the ligation method, the barcodes are prefixed with NB, where the rapid barcodes are prefixed with BC. Change in the commands accordingly</p>
<p>move to the medaka folder</p>
<pre><code>cd ../medaka
</code></pre>
<p><strong>ARTIC 400bp primers</strong></p>
<pre><code>for i in "01 0013" "02 0014"; do a=( $i ); mkdir midnight_R_nCoV_${a[1]}_${RUN}_NB${a[0]}_medaka; cd midnight_R_nCoV_${a[1]}_${RUN}_NB${a[0]}_medaka; artic minion --minimap2 --medaka --normalise 200 --threads 8 --scheme-directory /data1/software/artic-ncov2019/primer_schemes --read-file ../../base/${RUN}_fastq_pass-NB${a[0]}.fastq nCoV-2019/V3 R_nCoV_${a[1]}; cd ../; done
</code></pre>
<p><strong>Midnight 1200bp primers</strong></p>
<pre><code>for i in "01 0013" "02 0014"; do a=( $i ); mkdir midnight_R_nCoV_${a[1]}_${RUN}_BC${a[0]}_medaka; cd midnight_R_nCoV_${a[1]}_${RUN}_BC${a[0]}_medaka; artic minion --minimap2 --medaka --normalise 200 --threads 8 --scheme-directory /data1/software/SARS-CoV-2_GTG/protocols/Midnight/schemes --read-file ../../base/${RUN}_fastq_pass-BC${a[0]}.fastq nCoV-2019/V1 R_nCoV_${a[1]}; cd ../; done
</code></pre>
<p><strong>Eden/Kirby 2500bp primers</strong></p>
<pre><code>for i in "01 0013" "02 0014"; do a=( $i ); mkdir midnight_R_nCoV_${a[1]}_${RUN}_BC${a[0]}_medaka; cd midnight_R_nCoV_${a[1]}_${RUN}_BC${a[0]}_medaka; artic minion --minimap2 --medaka --normalise 200 --threads 8 --scheme-directory /data1/software/SARS-CoV-2_GTG/protocols/Kirby/schemes --read-file ../../base/${RUN}_fastq_pass-BC${a[0]}.fastq nCoV-2019/V1 R_nCoV_${a[1]}; cd ../; done
</code></pre>
<p><strong>Example of medaka only output structure</strong></p>
<pre><code>PKRP008209/.
├── base
├── medaka
│   ├── kirby_nCoV_1905_PKRP008209_BC09_medaka
│   ├── kirby_nCoV_1906_PKRP008209_BC10_medaka
│   ├── kirby_nCoV_1907_PKRP008209_BC11_medaka
│   └── kirby_nCoV_1908_PKRP008209_BC12_medaka
└── PKRP008209
└── 1905-8
    └── 20210113_0459_1C_PAF26262_d0402aba
        ├── fastq_fail
        └── fastq_pass
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../tute/" class="btn btn-neutral float-right" title="Tutorial">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../realtime/" class="btn btn-neutral" title="Real-time monitoring"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../realtime/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../tute/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
